### PR TITLE
Remove IBC and Ngen related structures and comments

### PR DESCRIPTION
**Pull request by davidwrighton**
- Remove all IBC comments from the vm portion of the codebase
- Remove `m_GenericTypeDefToCanonMethodTableMap` and `m_MethodDefToPropertyInfoMap` which were only filled in by NGen
- Remove a variety of flags enums which are never used
- Remove infra for testing Zap and IBC  scenarios

### CODE REVIEW

File: src/coreclr/vm/assembly.cpp [hunk 1]
```diff
@@ -1640,119 +1640,6 @@ BOOL Assembly::GetResource(LPCSTR szName, DWORD *cbResource,
...
 }
 
-#ifdef FEATURE_READYTORUN
-BOOL Assembly::IsInstrumented()
-{
-    STATIC_CONTRACT_THROWS;
-    STATIC_CONTRACT_GC_TRIGGERS;
-    STATIC_CONTRACT_FAULT;
-
-    // This will set the value of m_isInstrumentedStatus by calling IsInstrumentedHelper()
```
Comment (elinor-fung) [inline src/coreclr/vm/assembly.cpp:1650]: Can we also remove `m_isInstrumentedStatus` and `IsInstrumentedStatus` now?

File: src/coreclr/vm/clsload.cpp [hunk 2]
```diff
@@ -769,19 +769,10 @@ TypeHandle ClassLoader::LoadConstructedTypeThrowing(TypeKey *pKey,
...
     // Lookup in the classes that this class loader knows about
 
-    if (pKey->HasInstantiation() && ClassLoader::IsTypicalSharedInstantiation(pKey->GetInstantiation()))
-    {
-        _ASSERTE(pKey->GetModule() == ComputeLoaderModule(pKey));
-        typeHnd = pKey->GetModule()->LookupFullyCanonicalInstantiation(pKey->GetTypeToken(), &existingLoadLevel);
-    }
-
-    if (typeHnd.IsNull())
+    typeHnd = LookupTypeHandleForTypeKey(pKey);
```
Comment (elinor-fung) [inline src/coreclr/vm/clsload.cpp:772]: Move the declaration of `typeHnd` to this line too?

File: src/coreclr/vm/clsload.cpp [hunk 3]
```diff
@@ -769,19 +769,10 @@ TypeHandle ClassLoader::LoadConstructedTypeThrowing(TypeKey *pKey,
...
+    if (!typeHnd.IsNull())
     {
-        typeHnd = LookupTypeHandleForTypeKey(pKey);
-        if (!typeHnd.IsNull())
-        {
-            existingLoadLevel = typeHnd.GetLoadLevel();
-        }
+        existingLoadLevel = typeHnd.GetLoadLevel();
```
Comment (elinor-fung) [inline src/coreclr/vm/clsload.cpp:775]: I think the if block below and the declaration of `existingLoadLevel` can be moved into this now.

**Review by elinor-fung — APPROVED** [review]

File: src/coreclr/vm/ceeload.cpp [hunk 4]
```diff
@@ -3439,7 +3418,6 @@ MethodDesc *Module::FindMethod(mdToken pMethod)
 
 //
 // GetPropertyInfoForMethodDef wraps the metadata function of the same name,
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/ceeload.cpp:3420]:
```suggestion
// GetPropertyInfoForMethodDef wraps the metadata function of the same name.
```

File: src/coreclr/vm/jitinterface.cpp [hunk 5]
```diff
@@ -5700,17 +5700,17 @@ unsigned CEEInfo::getClassDomainID (CORINFO_CLASS_HANDLE clsHnd,
 
 //---------------------------------------------------------------------------------------
 //
-// Used by the JIT to determine whether the profiler or IBC is tracking object
+// Used by the JIT to determine whether the profiler  is tracking object
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/jitinterface.cpp:5703]:
```suggestion
// Used by the JIT to determine whether the profiler is tracking object
```

File: src/coreclr/vm/jitinterface.cpp [hunk 6]
```diff
@@ -5700,17 +5700,17 @@ unsigned CEEInfo::getClassDomainID (CORINFO_CLASS_HANDLE clsHnd,
...
 // allocations
 //
 // Return Value:
-//    bool indicating whether the profiler or IBC is tracking object allocations
+//    bool indicating whether the profiler  is tracking object allocations
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/jitinterface.cpp:5707]:
```suggestion
//    bool indicating whether the profiler is tracking object allocations
```

File: src/coreclr/vm/zapsig.cpp [hunk 7]
```diff
@@ -1250,10 +1246,8 @@ BOOL ZapSig::EncodeMethod(
 
         if (pTypeHandleModule != pInfoModule)
         {
-            // During IBC profiling this calls
-            //     code:Module.EncodeModuleHelper
-            // During ngen this calls
-            //     code:ZapImportTable.EncodeModuleHelper)
+        // During multicorejit this calls
+        //     code:MulticoreJitManager.EncodeModuleHelper
             //
```
Comment (AaronRobinsonMSFT) [inline src/coreclr/vm/zapsig.cpp:1251]:
```suggestion
```

**Review by AaronRobinsonMSFT — APPROVED** [review]

File: src/coreclr/vm/assembly.cpp (hunk 1 above)
Comment (davidwrighton) [reply src/coreclr/vm/assembly.cpp:1650]: Yes. I intended to to that, and got... distracted deleting other stuff...

File: src/coreclr/vm/clsload.cpp (hunk 2 above)
Comment (davidwrighton) [reply src/coreclr/vm/clsload.cpp:772]: sure.

File: src/coreclr/vm/clsload.cpp (hunk 3 above)
Comment (davidwrighton) [reply src/coreclr/vm/clsload.cpp:775]: We don't even need a declaration for `existingLoadLevel`.

File: src/coreclr/inc/clrconfigvalues.h [hunk 8]
```diff
@@ -662,8 +662,6 @@ RETAIL_CONFIG_DWORD_INFO(INTERNAL_EnableDumpOnSigTerm, W("EnableDumpOnSigTerm"),
 ///
 /// Zap
 ///
-RETAIL_CONFIG_STRING_INFO(INTERNAL_ZapBBInstr, W("ZapBBInstr"), "")
-RETAIL_CONFIG_STRING_INFO(EXTERNAL_ZapBBInstrDir, W("ZapBBInstrDir"), "")
 RETAIL_CONFIG_DWORD_INFO(EXTERNAL_ZapDisable, W("ZapDisable"), 0, "")
```
Comment (EgorBo) [inline src/coreclr/inc/clrconfigvalues.h:665]: `ZapDisabled` is kept to maintain compatibility and is the same as ReadyToRun=0?

**Pull request merged**